src/benchmarklib/benchmark_utils.cpp

```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
       enable_visualization(enable_visualization),
       out(out) {}
 
+BenchmarkConfig BenchmarkConfig::get_default_config() { return BenchmarkConfig(); }
+
+bool CLIConfigParser::cli_has_json_config(const int argc, char** argv) {
+  const auto has_json = argc > 1 && boost::algorithm::ends_with(argv[1], ".json");
+  if (has_json && argc > 2) {
+      std::cout << "Passed multiple args with a json config. All CLI args will be ignored...";
+  }
+
+  return has_json;
+}
+
+nlohmann::json CLIConfigParser::config_file_to_json(const std::string& json_file_str) {
```

Comment:
This does not sound right. The config file *is* a json. Maybe `parse_json_config_file` or something alike?